sqfentity/lib/sqfentity.dart

```diff
@@ -336,13 +351,17 @@ class SqfEntityProvider extends SqfEntityModelBase {
     }
   }
 
-  Future<int?> insert<T extends TableBase>(T obj, bool ignoreBatch) async {
+  Future<int?> insert<T extends TableBase>(T obj, bool ignoreBatch,
+      [bool useHook = true]) async {
     try {
       /// Leave it in this format for Throw to stay in this catch
       final res = await insertOrThrow(obj, ignoreBatch);
+
       if (_dbModel!.postSaveAction != null) {
-        final record = obj.toMap(forQuery: true);
-        await _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
+        if (useHook) {
+          final record = obj.toMap(forQuery: true);
+          await _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
+        }
       }

```

Comment:
Same here. One line condition statement should suffice `if (_dbModel!.postSaveAction != null && useHook) {`